src/hooks/useNotifications.tsx

```diff
@@ -85,6 +95,11 @@ const useNotificationsContext = () => {
 
   // Check for version changes
   useEffect(() => {
+    const MAX_NOTIFICATIONS = 50;
+    if (Object.keys(notifications).length > MAX_NOTIFICATIONS) {
```

Comment:
suggestion:
I think instead of checking for a max count, we should check for if there are any AbacusGenerated notifications and then clear if there are, something like:

```
const clearAbacusGeneratedNotifications = useCallback(() => {
  const originalEntries = Object.entries(notifications);
  const filteredEntries = originalEntries.filter(
    ([, value]) => value.type !== NotificationType.AbacusGenerated
  );

  // Only update if the number of notifications has changed
  if (filteredEntries.length !== originalEntries.length) {
    const newNotifications = Object.fromEntries(
      filteredEntries.map(([key, value]) => [getKey(value), value])
    );
    setNotifications(newNotifications);
  }
}, [notifications]);
```